cmd/segment-reaper/observer.go

```diff
@@ -143,11 +168,87 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
```

Comment:
This says that if an object has an `l` segment and is not supposed to have any `s` segments then any good sequence of `s` segments makes is a correct object... Which sounds wrong to me.

In my opinion, if an object has an `l` segment and is not supposed to have any `s` segments then this is always a good object and we should not report it.

But! If we find any `s` segments for this same object then these are zombie segments - perhaps leftover from a previous version of this object. We should report the `s` segments, but not the `l` segment.

Reply:
In the more generic case, if the object has an `l` segment and is supposed to have X number of `s` segments, then we should not consider it broken if we find all the `l`, `s1`, ..., and `sX` segments. But any segments beyond `sX` should be reported as zombies.

Reply:
> This says that if an object has an `l` segment and is not supposed to have any `s` segments then any good sequence of `s` segments makes is a correct object... Which sounds wrong to me.

This part means that if `l` segment doesn't have unencrypted number of segments (`object.expectedNumberOfSegments`) then we need to check sequence. For object with only `l` segment bitmask will be `000000...` and `IsSequence` will return `true`.

Reply:
Yes, I agree. I was mentioning this in the second comment.

Reply:
> In the more generic case, if the object has an `l` segment and is supposed to have X number of `s` segments, then we should not consider it broken if we find all the `l`, `s1`, ..., and `sX` segments. But any segments beyond `sX` should be reported as zombies.

I agree. I was mentioning this in my second response.